site-manage/datemanager/impl/src/java/org/sakaiproject/datemanager/impl/DateManagerServiceImpl.java

```diff
@@ -1248,12 +1248,19 @@ public void updateAnnouncements(DateManagerValidation announcementValidate) thro
 	@Override
 	public JSONArray getLessonsForContext(String siteId) {
 		JSONArray jsonLessons = new JSONArray();
-		String url = getUrlForTool(DateManagerConstants.COMMON_ID_LESSONS);
-		List<SimplePageItem> items = simplePageToolDao.findItemsInSite(siteId);
+		jsonLessons = addAllSubpages(siteId, null, jsonLessons, "false");
+		return jsonLessons;
+	}
+
+	private JSONArray addAllSubpages(String siteId, Long pageId, JSONArray jsonLessons, String extraInfo) {
+		List<SimplePageItem> items = siteId!=null?simplePageToolDao.findItemsInSite(siteId):simplePageToolDao.findItemsOnPage(pageId);
```

Comment:
The previous code loaded all the items in the site before calling the recursive method can we go back to that as it was clearer and it avoids the potential of that being called more than once.

If not then put it in an if statement at the very top of the method with an explanation.

Reply:
I've made a new version easy to read !!